src/lib/core/types/AnalysisStore.ts

```diff
@@ -0,0 +1,32 @@
+import { AnalysisClient } from '../api/analysis-api';
+import { Analysis } from './analysis';
+
+interface SubscribeCallback {
+  (analysis: Analysis): void;
+}
+
+export class AnalysisStore {
```

Comment:
Looking forward to seeing how this gets used!

Reply:
Oh no! I did not mean to include that :facepalm: